pallets/amm/src/mock.rs

```diff
@@ -1,25 +1,31 @@
 use crate as pallet_amm;
 
 use codec::{Decode, Encode, MaxEncodedLen};
-use frame_support::traits::Contains;
-use frame_support::{parameter_types, traits::GenesisBuild, PalletId};
+use frame_benchmarking::frame_support::traits::tokens::{DepositConsequence, WithdrawConsequence};
+use frame_support::traits::fungible::{
+    Inspect as FungibleInspect, Mutate as FungibleMutate, Transfer as FungibleTransfer,
+};
+use frame_support::traits::fungibles::{Inspect, Mutate, Transfer};
+use frame_support::{parameter_types, PalletId};
 use frame_system as system;
-use orml_traits::parameter_type_with_key;
-use primitives::{Amount, Balance, CurrencyId, TokenSymbol};
+use frame_system::EnsureRoot;
+use primitives::currency::CurrencyOrAsset;
+use primitives::{Balance, TokenSymbol};
 #[cfg(feature = "std")]
 use serde::{Deserialize, Serialize};
 use sp_core::H256;
-use sp_runtime::traits::AccountIdConversion;
+use sp_runtime::DispatchError;
+use sp_runtime::DispatchResult;
 pub use sp_runtime::Perbill;
 use sp_runtime::{
     testing::Header,
     traits::{BlakeTwo256, IdentityLookup},
     RuntimeDebug,
 };
-
-pub const DOT: CurrencyId = CurrencyId::Token(TokenSymbol::DOT);
-pub const XDOT: CurrencyId = CurrencyId::Token(TokenSymbol::xDOT);
-pub const NATIVE: CurrencyId = CurrencyId::Token(TokenSymbol::HKO);
+use std::marker::PhantomData;
+pub const DOT: CurrencyOrAsset = CurrencyOrAsset::NativeCurrency(TokenSymbol::DOT);
```

Comment:
Why native token is also a TokenSymbol::DOT ? I think a native token is like HKO, and DOT is a kind of asset management by pallet assets.

Reply:
What do you suggest @ETeissonniere ? Shall we rename NativeCurrency or add one more field in `CurrencyOrAsset`

Reply:
My suggestion is that NativeCurrency is an element of enum without TokenSymbol type. It already modify by @GopherJ in this pr. https://github.com/parallel-finance/parallel/pull/607/files#diff-9a08c969144a056406ced2d9dde2f410e0135639d066f770d14b702ebf909a80R113